libraries/helpers.rb

```diff
@@ -1,67 +1,248 @@
-require 'shellwords'
-
 module MysqlCookbook
-  module Helpers
-    include Chef::DSL::IncludeRecipe
+  module HelpersBase
+    require 'shellwords'
 
-    def base_dir
-      prefix_dir || '/usr'
+    def el5?
+      return true if node['platform_family'] == 'rhel' && node['platform_version'].to_i == 5
+      false
     end
 
-    def configure_package_repositories
-      # we need to enable the yum-mysql-community repository to get packages
-      return unless %w(rhel fedora).include? node['platform_family']
-      case parsed_version
-      when '5.5'
-        # Prefer packages from native repos
-        return if node['platform_family'] == 'rhel' && node['platform_version'].to_i == 5
-        return if node['platform_family'] == 'fedora'
-        include_recipe('yum-mysql-community::mysql55')
-      when '5.6'
-        include_recipe('yum-mysql-community::mysql56')
-      when '5.7'
-        include_recipe('yum-mysql-community::mysql57')
-      end
+    def el6?
+      return true if node['platform_family'] == 'rhel' && node['platform_version'].to_i == 6
+      false
+    end
+
+    def el7?
+      return true if node['platform_family'] == 'rhel' && node['platform_version'].to_i == 7
+      false
+    end
+
+    def fc23?
+      return true if node['platform'] == 'fedora' && node['platform_version'].to_i == 23
+      false
+    end
+
+    def fc24?
```

Comment:
It would be great to kill the use of FCXYZ helpers and just have a fedora? helper. Otherwise we have to upgrade this thing every time a new fedora comes out